File: app/src/main/java/com/numero/sojodia/model/BusDataFile.java
package com.numero.sojodia.model;

public enum BusDataFile {

    TK_TO_KUTC(
            "TkToKutc.csv"
    ),

    TND_TO_KUTC(
            "TndToKutc.csv"
    ),

    KUTC_TO_TK(
            "KutcToTk.csv"
    ),

    KUTC_TO_TND(
            "KutcToTnd.csv"
    );

    private static final String baseUrl = "https://raw.githubusercontent.com/NUmeroAndDev/SojoDia-BusDate/master/%s";

    private String url;
    private String fileName;

    BusDataFile(String fileName) {
        this.url = String.format(baseUrl, fileName);
        this.fileName = fileName;
    }

    public String getUrl() {
        return url;
    }

    public String getFileName() {
        return fileName;
    }
}

File: app/src/main/kotlin/com/numero/sojodia/model/BusDataFile.kt
package com.numero.sojodia.model

import com.numero.sojodia.BuildConfig

enum class BusDataFile(val fileName: String) {

    TK_TO_KUTC(
            "TkToKutc.csv"
    ),

    TND_TO_KUTC(
            "TndToKutc.csv"
    ),

    KUTC_TO_TK(
            "KutcToTk.csv"
    ),

    KUTC_TO_TND(
            "KutcToTnd.csv"
    );

    val url: String = BuildConfig.BUS_FILE_URL.format(fileName)

}
